node/sc/bridgepool/sorted_map_list.go

```diff
@@ -203,6 +220,27 @@ func (m *ItemSortedMap) Remove(nonce uint64) bool {
 	return true
 }
 
+// Forward removes all items from the map with a nonce lower than the
+// provided threshold. Every removed transaction is returned for any post-removal
+// maintenance.
+func (m *ItemSortedMap) Forward(threshold uint64) items {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	var removed items
+
+	// Pop off heap items until the threshold is reached
+	for m.index.Len() > 0 && (*m.index)[0] < threshold {
+		nonce := heap.Pop(m.index).(uint64)
+		removed = append(removed, m.items[nonce])
```

Comment:
How about removing the `items` which is not used currently?

Reply:
Returned items can be used. So not to remove is better as we talked.